provider/cmd/pulumi-resource-aws-native/schema.json

```diff
@@ -173204,6 +173204,13 @@
                 "outputs"
             ],
             "inputProperties": {
+                "createOnly": {
+                    "type": "array",
+                    "items": {
+                        "type": "string"
+                    },
+                    "description": "Property names as defined by `createOnlyProperties` in the CloudFormation schema. Create-only properties can't be set during updates, so will not be included in patches even if they are also marked as write-only, and will cause an error if attempted to be updated.\nIn the CloudFormation schema these are fully qualified property paths (e.g. `/properties/AccessToken`) whereas here we only include the top-level property name (e.g. `AccessToken`)."
```

Comment:
Yup, replace on changes would have to be set too. Though the failure mode would just be the service saying you can't update that field which should be quite straightforward to fix.

Unfortunately write only properties need to be known for both the patch operation but also for the read operation. So if the user removes it from the write only, it'll fix the update but then break the reading. In practice I expect these won't be used much at all as they really are wedge cases and I expect third party resources should be less weird. That said, if we want to ensure accessibility for all resources, I can't see a way of avoiding this.

Reply:
It would be nice if we could just access replaceOnChanges from the provider as that would be sufficient to document that they have to set this, but that's not currently possible as far as I know.

Reply:
Bummer that we don't have access to the `replaceOnChanges` option. We're kind of stuck in the other direction too it seems since we can't generate replaceOnChanges into the SDK for ExtensionResouces like we do for the others. I don't feel like this is a great place to leave it, but don't see a lot of good alternatives. At minimum, the comment on this parameter should note that setting `createOnly` isn't sufficient and users will also need to set the `replaceOnChanges` option.